promtimer/cbstats.py

```diff
@@ -429,14 +435,36 @@ def get_stats_sources(cluster, user, password):
                 if host is None:
                     host = '127.0.0.1'
                 services = node['services']
-                port = services['mgmt']
-                source = ServerNode(host, port, user, password)
+                port = services['mgmtSSL'] if secure else services['mgmt']
+                source = ServerNode(host, port, user, password, secure)
                 result.append(source)
             return result
         except OSError as err:
             logging.error('error: can\'t access cluster: {}'.format(err))
             return []
 
+    @staticmethod
+    def get_stats_sources_from_nodes(nodes, user, password):
+        result = []
+        for node in nodes:
+            secure = re.match('https://', node, re.IGNORECASE)
```

Comment:
Nitpicking: There are better ways than regex to do this.

Reply:
Parsing url and using scheme now